### PR TITLE
Sync quarkus3 branch with upstream/main 

### PR DESCRIPTION
**Pull request by Pepo48**
JFTR here's the full list of conflicts for this iteration:
```
# Conflicts:
#    model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaModelCriteriaBuilder.java
#    model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/dialect/JsonbPostgreSQL95Dialect.java
#    quarkus/deployment/pom.xml
#    quarkus/pom.xml
#    quarkus/runtime/pom.xml
#    testsuite/integration-arquillian/tests/base/pom.xml
#    testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientPoliciesTest.java
#    testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientRegistrationTest.java
```

### CODE REVIEW

**Review by pedroigor — APPROVED** [review]

File: pom.xml
Comment (vmuzikar) [inline pom.xml:97]: Is this expected? Is it caused by the migration script?

File: quarkus/tests/integration/src/test-providers/resources/com/acme/provider/legacy/jpa/entity/persistence.xml
Comment (vmuzikar) [inline quarkus/tests/integration/src/test-providers/resources/com/acme/provider/legacy/jpa/entity/persistence.xml:28]: Is this expected? Is it caused by the migration script?

**Review by vmuzikar — COMMENTED** [review]
@Pepo48 Nice work, LGTM.

Before merging I think it'd be nice to also get a review from @mabartos and someone from @keycloak/storage-x.

File: pom.xml
Comment (Pepo48) [reply pom.xml:97]: @vmuzikar yes, it is.

File: quarkus/tests/integration/src/test-providers/resources/com/acme/provider/legacy/jpa/entity/persistence.xml
Comment (Pepo48) [reply quarkus/tests/integration/src/test-providers/resources/com/acme/provider/legacy/jpa/entity/persistence.xml:28]: @vmuzikar yes, it is caused by the script.

File: model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/dialect/JsonbPostgreSQL95Dialect.java
Comment (Pepo48) [inline model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/dialect/JsonbPostgreSQL95Dialect.java:30]: @mabartos @keycloak/storage-x during the process I noticed that all the `registerFunction()` methods in the constructor had been commented out. There was a conflict, which I resolved by adding the new method (L30) and commenting it out right afterwards. 

I suppose that for now this is expected, just wanted to specifically point it out as a reminder with regards to follow-ups.

Comment (vramik) [reply model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/dialect/JsonbPostgreSQL95Dialect.java:30]: Thanks @Pepo48, as you say. It's expected and your solution is correct.

**Review by mabartos — APPROVED** [review]
@Pepo48 Thanks! LGTM. 

Be aware the migration script is not still perfect and contains some inaccuracies, which is also the point where you needed to revert some changes and it probably caused some tediousness :))

I've tried to address these inaccuracies here[1]. I'll prepare the PRs for the Quarkus repo.

[1]  https://github.com/keycloak/keycloak/issues/16247

**Comment by hmlnarik** [comment]
@Pepo48 Can you please provider the list of conflicts?

**Comment by Pepo48** [comment]
> @Pepo48 Can you please provider the list of conflicts?

@hmlnarik here you go:
```
# Conflicts:
#	model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaModelCriteriaBuilder.java
#	model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/dialect/JsonbPostgreSQL95Dialect.java
#	testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientPoliciesTest.java
#	testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientRegistrationTest.java
```

(it was already mentioned in the commit message of the merge commit (the first one) but I'll make it more transparent next time)

**Review by hmlnarik — APPROVED** [review]
> > @Pepo48 Can you please provider the list of conflicts?
> 
> @hmlnarik here you go:
> 
> ```
> # Conflicts:
> #	model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaModelCriteriaBuilder.java
> #	model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/hibernate/dialect/JsonbPostgreSQL95Dialect.java
> #	testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientPoliciesTest.java
> #	testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientRegistrationTest.java
> ```
> 
> (it was already mentioned in the commit message of the merge commit (the first one) but I'll make it more transparent next time)

Thank you! Would be perfect to have this information readily available at the top of the PR.

LGTM

**Comment by Pepo48** [comment]
As per @hmlnarik request I re-synced the branch with upstream/main and added a new merge commit.
Here is the list of conflicts:
```
# Conflicts:
#    quarkus/deployment/pom.xml
#    quarkus/pom.xml
#    quarkus/runtime/pom.xml
#    testsuite/integration-arquillian/tests/base/pom.xml
```

I ran the Quarkus migration script again - this time no changes were necessary.
I built and ran the distribution just to be completely sure and everything worked as expected.

**Review by mabartos — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]
@Pepo48 Thanks for the update!

**Comment by vmuzikar** [comment]
There still seem to be some conflicts preventing the merged. I tried to resolve them automatically (GitHub added another merge commit) but it didn't help. I think we need to rebase on current `upstream/quarkus3` branch manually.

**Review by mabartos — APPROVED** [review]

**Comment by hmlnarik** [comment]
Superseded by rebasing

**Pull request closed without merging**
